kernel/src/sched.rs

```diff
@@ -61,16 +61,37 @@ impl Kernel {
     }
 
     /// Something was scheduled for a process, so there is more work to do.
+    ///
+    /// This is only exposed in the core kernel crate.
     pub(crate) fn increment_work(&self) {
         self.work.increment();
     }
 
+    /// Something was scheduled for a process, so there is more work to do.
+    ///
+    /// This is exposed publicly, but restricted with a capability. The intent
+    /// is that external implementations of `ProcessType` need to be able to
+    /// indicate there is more process work to do.
+    pub fn increment_work_public(&self, _capability: &dyn capabilities::MainLoopCapability) {
```

Comment:
Would it make more sense for this to just take in `ExternalProcessCapability`? It doesn't make a lot of sense to me to give some process type the ability to start the main loop, which is what this capability was previously used for.

For reference, [here](https://github.com/jon-flatley/tock/blob/124383bbc7dbc0916800359adc67bbd73da8f48d/host-testing/src/process.rs#L266) is where it is called in the host side code.

Most current uses of this function in Tock come from within `process.rs` anyway, as opposed to `sched.rs`

Reply:
I can get behind this idea.